runtime/ms-rest-azure/lib/filters/rpRegistrationFilter.js

```diff
@@ -50,6 +60,35 @@ exports.create = function create(retryTimeoutInSec) {
 };
 
 /**
+ * Reuses the headers of the original request and url (if specified).
+ * @param {object} originalRequest The original request
+ * @param {boolean} reuseUrlToo Should the url from the original request be reused as well. Default false.
+ * @returns {object} reqOptions - A new request object with desired headers.
+ */
+function getRequestEssentials(originalRequest, reuseUrlToo) {

```

Comment:
consider:  you should be able to know what the function does by reading its name - something like `prepareRequestHeader` makes a better choice here.